sdk/core/core-http/test/policies/bearerTokenAuthenticationPolicyTests.ts

```diff
@@ -88,6 +71,66 @@ describe("BearerTokenAuthenticationPolicy", function() {
     assert.strictEqual(credentialToTest.authCount, 2);
   });
 
+  it("refreshes the token on initial request", async () => {
+    const expiresOn = Date.now() + 1000 * 60; // One minute later.
+    const credential = new MockRefreshAzureCredential(expiresOn);
+    const request = createRequest();
+    const policy = createBearerTokenPolicy("test-scope", credential);
+
+    await policy.sendRequest(request);
+    assert.strictEqual(credential.authCount, 1);
+  });
+
+  it("refreshes the token again only once it expired", async () => {
+    const expireDelayMs = 5000;
+    let tokenExpiration = Date.now() + expireDelayMs;
+    const credential = new MockRefreshAzureCredential(tokenExpiration);
+    const request = createRequest();
+    const policy = createBearerTokenPolicy("test-scope", credential);
+
+    // The token is cached and remains cached for a bit.
+    await policy.sendRequest(request);
+    await policy.sendRequest(request);
+    assert.strictEqual(credential.authCount, 1);
+
+    // The token will remain cached until tokenExpiration - testTokenRefreshBufferMs, so in (5000 - 1000) milliseconds.
+
+    // For safe measure, we test the token is still cached a second earlier than the refresh expectation.
+    await delay(expireDelayMs - testTokenExpiresThisEarlier - 1000);
```

Comment:
sorry to do it to you, but can we still use sinon to avoid long waits here?

Reply:
Can do! Thank you for the impulse on this direction. Sinon time fancy stuff is really amazing!